packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -140,6 +153,27 @@ def _GenerateSshKey(self, key_type, key_dest):
 
     file_utils.SetPermissions(key_dest, mode=0o600)
     file_utils.SetPermissions('%s.pub' % key_dest, mode=0o644)
+    with open('%s.pub' % key_dest, 'r') as pk:
+        key_data = pk.read()
+
+    key_values = key_data.split()
+    if len(key_values) < 2:
+        return
+    else:
+        return key_values[0], key_values[1]
+
+  def _WriteHostKeyToGuestAttributes(self, key_type, key_value):
+    """Write a host key to guest attributes, ignoring errors."""
+    headers = {'Metadata-Flavor': 'Google'}
+    url = GUEST_ATTRIBUTES_URL + key_type
+    req = PutRequest(url, key_value, headers)
+    try:
+      response = urlrequest.urlopen(req)
+      self.logger.debug(response)
+      self.logger.info('Wrote %s host key to guest attributes.' % key_type)
```

Comment:
For `logger.info` you should use a comma for variable substitution arguments.

Reply:
Done.